### PR TITLE
Updated flow-bin to the latest version (to date)

### PR DESCRIPTION
**Pull request by mondoreale**
I went from `0.75.0` all the way to the most recent `0.96.0`. Did a bunch of stops down the road to make the migration self-explanatory. Each commit is a version that required some doing to get `No errors!`.

Hope it's clear and you guys like it. Cheers!

### CODE REVIEW

File: app/src/shared/components/ModalDialog/index.jsx [hunk 1]
```diff
@@ -11,12 +11,15 @@ import styles from './modalDialog.pcss'
 
 export type Props = {
     onClose: () => void,
-    children: Node,
     className?: string,
     backdropClassName?: string,
 }
 
-const ModalDialog = ({ onClose, children, className, backdropClassName }: Props) => (
+export type InternalProps = Props & {
```
Comment (mondoreale) [inline app/src/shared/components/ModalDialog/index.jsx]: FYI, this is the issue I raised up on Slack, i.e. `{ key?: type } & { key: type }`. Flow gives a very sketchy error message for it that does not help figuring out wtf.

File: app/src/shared/utils/withErrorBoundary.jsx [hunk 2]
```diff
@@ -26,10 +26,10 @@ const withErrorBoundary = (ErrorComponent: ComponentType<any>) => (
                 }
             }
 
-            componentDidCatch(error: Error, errorInfo: string) {
+            componentDidCatch(error: Error, extra: any) {
                 console.error(error)
                 Raven.captureException(error, {
-                    extra: errorInfo,
+                    extra,
```
Comment (timoxley) [inline app/src/shared/utils/withErrorBoundary.jsx]: 👍

File: app/src/marketplace/components/Nav/index.jsx [hunk 3]
```diff
@@ -67,14 +67,14 @@ class Nav extends React.Component<Props> {
                         <Translate value="general.streams" />
                     </Link>
                 </NavDropdown>
-                <NavDivider />
+                {NavDivider}
```
[the remark this answers is not in the document]
Comment (mondoreale) [reply app/src/marketplace/components/Nav/index.jsx]: Well, flow complained about something there – that's how I noticed it – but I can try to trick it into believing it's all fine (flow is a faisty believer sometimes). And yeah, it does look kinda odd when surrounded with differently flavoured code.

[the remark this answers is not in the document]
Comment (mondoreale) [reply app/src/marketplace/components/Nav/index.jsx]: Ok. It doesn't complain and `NavDivider` can simply be used just like other stuff. Weird but it's fine.

Flow used to complain on `const NavDivider = () => …` and not on `const NavDivider = (props: {}) => …`, FYI.

**Pull request merged**